File: src/lib/agents/tech-agent.ts
/**
 * Tech Agent
 * 
 * Specializes in technology companies, AI, software, hardware, and innovation.
 */

import { BaseAgent, type AgentConfig } from './base-agent';

const TECH_CONFIG: AgentConfig = {
  name: 'Tech Agent',
  section: 'tech',
  articlesPerRun: 2,
  author: 'Innovation & Technology',
  toneGuide: `
    - Tech-savvy but accessible
    - Cut through hype to find substance
    - Focus on business and societal implications
    - Explain technical concepts simply
    - Healthy skepticism toward PR claims
  `,
  styleGuide: `
    - Lead with the impact, not the technology itself
    - Explain technical concepts in plain English
    - Include specific metrics: users, revenue, performance
    - Quote technologists and industry analysts
    - Note competitive dynamics (Apple vs Google, etc.)
    - Address privacy, security, and ethical concerns
    - Connect to broader tech industry trends
  `,
};

export class TechAgent extends BaseAgent {
  constructor() {
    super(TECH_CONFIG);
  }

  protected getWritingSystemPrompt(): string {
    return `You are a senior technology correspondent at The Wire Journal, covering Silicon Valley and the global tech industry.

${super.getWritingSystemPrompt()}

TECH NEWS GUIDELINES:
- Make complex technology understandable
- Focus on business impact, not just cool features
- Be skeptical of company claims - verify with data
- Address regulatory and antitrust implications
- Note AI/automation impacts where relevant
- Include user/consumer perspective
- Explain what this means for the industry
- Cover both innovation and disruption angles`;
  }
}

let techAgent: TechAgent | null = null;

export function getTechAgent(): TechAgent {
  if (!techAgent) {
    techAgent = new TechAgent();
  }
  return techAgent;
}

File: backend/agents/tech_agent.py
"""
Tech Agent - Innovation & Technology
Covers technology companies, AI, software, hardware, innovation
"""
from agents.base_agent import BaseAgent, AgentConfig


class TechAgent(BaseAgent):
    """Agent for technology news coverage"""
    
    def __init__(self):
        config = AgentConfig(
            name='tech',
            section='tech',
            articles_per_run=2,
            author='Innovation & Technology',
            tone_guide='Forward-looking, technically informed, accessible.',
            style_guide="""
- Lead with innovation and disruption angles
- Explain technical concepts clearly
- Include expert and analyst perspectives
- Discuss broader implications (privacy, regulation, competition)
- Note adoption trends and market reception
- Balance hype with realistic assessment
            """
        )
        super().__init__(config)

